six/modules/c++/six.sicd/tests/test_compare_sicd_meshes.cpp

```diff
@@ -0,0 +1,363 @@
+                  << baselineSicdPathname << " size = " << baselineData.size()
+                  << ", " << testSicdPathname << " size = " << testData.size()
+                  << std::endl;
+
+        return false;
+    }
+
+    if (baselineData.size() != dims.area())
+    {
+        std::cerr << "Failure - Dimensions (row = " << dims.row << ", col = "
+                  << dims.col << ") do not match num elements in mesh ("
+                  << baselineData.size() << " elements)." << std::endl;
+        return false;
+    }
+
+    for (size_t row = 0; row < dims.row; ++row)
+    {
+        for (size_t col = 0; col < dims.col; ++col)
+        {
+            const size_t idx = row * dims.col + col;
+            if (baselineData[idx] != testData[idx])
+            {
+                std::cerr << "Failure - Values do not match at row = "
+                          << row << ", col = " << col << ": "
+                          << baselineSicdPathname << " value = "
+                          << baselineData[idx] << ", " << testSicdPathname
+                          << " value = " << testData[idx] << std::endl;
+                return false;
+            }
+        }
+    }
+
+    return true;
+}
+
+bool comparePlanarCoordinateMeshes(
+        const std::string& baselineSicdPathname,
+        const six::sicd::PlanarCoordinateMesh& baselineMesh,
+        const std::string& testSicdPathname,
+        const six::sicd::PlanarCoordinateMesh& testMesh)
+{
+    if (!compareMeshDims(baselineSicdPathname,
+                         baselineMesh,
+                         testSicdPathname,
+                         testMesh))
+    {
+        return false;
+    }
+
+    std::cout << "Comparing X coordinates..." << std::endl;
+    if (!compareMeshData(baselineSicdPathname,
+                         baselineMesh.getX(),
+                         testSicdPathname,
+                         testMesh.getX(),
+                         testMesh.getMeshDims()))
+    {
+        return false;
+    }
+
+    std::cout << "Comparing Y coordinates..." << std::endl;
+    if (!compareMeshData(baselineSicdPathname,
+                         baselineMesh.getY(),
+                         testSicdPathname,
+                         testMesh.getY(),
+                         testMesh.getMeshDims()))
+    {
+        return false;
+    }
+
+    return true;
+}
+
+bool compareNoiseMeshes(const std::string& baselineSicdPathname,
+                        const six::sicd::NoiseMesh& baselineMesh,
+                        const std::string& testSicdPathname,
+                        const six::sicd::NoiseMesh& testMesh)
+{
+
+    if (!comparePlanarCoordinateMeshes(baselineSicdPathname,
+                                       baselineMesh,
+                                       testSicdPathname,
+                                       testMesh))
+    {
+        return false;
+    }
+
+    std::cout << "Comparing azimuth ambiguity noise data..." << std::endl;
```

Comment:
Comparisons themselves all look good... but it might be useful to know, even if the az ambiguity noise data doesn't match, if the combined noise data matches.  So, throughout, always do all the comparisons (unless the meshes are a different size), and hold onto a true/false return code that's false if any of the comparisons fail.